### PR TITLE
Fix: update lesson-note--variations styles a11y

### LINKED ISSUE #4045: Update lesson note styles
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this issue follows the `brief description of request` format, e.g. `Add dark mode to website`

The following checkbox is OPTIONAL:

-   [ ] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**

Currently the lesson-note classe(es) have styling for a default (gold), info/tip (green), and warning (red). We should really update and add onto these styles so that the colors match the intention better, as well add icons for heading levels beyond `h4`

**2. Acceptance Criteria:**

- [ ] Update the styles so that:
  - there is a top and bottom margin for a lesson-note (we could use the same amount as the padding)
  - the `lesson-note--tip` class uses blue as the color; green is synonymous with "success" or "complete"
  - the `lesson-note--warning` class uses yellow or orange as the color; red is synonymous with "error" or "critical". This would also require updating the icon to a proper "warning" icon, such as the [Font Awesome exclamation triangle](https://fontawesome.com/v4/icon/exclamation-triangle)
  - This _may_ require updating the default lesson note color from Odin gold... it depends how close it is to the warning color
- [ ] Add a `lesson-note--critical` class that uses the current `lesson-note--warning` style. This styling should be used for things that are "critical" for users to read, a step above a simple "warning" 
- [ ] We should render the icon even if there is no heading, as the color alone won't be enough for users to convey the meaning if they suffer from any form of color blindness
  - One thing we could do is render a larger icon if there is no heading, otherwise render an icon that inherits the heading font size (this should also cover rendering the icon for *any* heading level in the note box, not just h4's):
  Lesson note icon without a heading

  (Doing this may negate the need to change the default lesson note color from Odin gold, since the icon will help convey the meaning)

**3. Additional Information:**

### PR DESCRIPTION
**Pull request by luuu-xu**
## Because

Current `lesson-note--warning` and `lesson-note--tip` etc has colours and icons not meeting good a11y standards.

## This PR

- `lesson-note` div now has vertical margins with `my-4`.
- `lesson-note--tip`'s colour is now **blue**
- `lesson-note--warning` now has **orange** colour instead of red, and uses a warning icon
- `lesson-note--critical` variation is added, with colour of **red**, and uses a critical icon
- All of lesson-note sections now will render an icon **in front of the first descendent** element (assuming `h1-h6`) with the appropriate font-size. 
- Otherwise if the section **only has one descendent** (assuming only `p`), the icon will be rendered at the top left corner, with an enlarged `font-size: 1.8rem`

## Issue

Closes #4045 

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

**Comment by luuu-xu** [comment]
@KevinMulhern Hi Kevin please review this PR as well. It's a fix of styles only.

**Comment by KevinMulhern** [comment]
Hey @luuu-xu, @thatblindgeye is best suited to review this one.

**Comment by thatblindgeye** [comment]
@luuu-xu I'll be taking a look at this tomorrow morning, looking forward to seeing how things look!

File: app/javascript/stylesheets/lesson-content.css [hunk 1]
```diff
@@ -24,33 +24,83 @@
...
+    font-size: 1.8rem;
+    font-weight: bold;
     content: "\f0eb";
-    @apply mr-2 text-green-600 dark:text-green-700;
+    @apply mr-2 text-blue-600 dark:text-blue-500;
   }
 
   .lesson-note--warning {
-    @apply border-red-700 dark:border-red-500;
+    @apply border-orange-700 dark:border-orange-600;
```
Comment (thatblindgeye) [inline app/javascript/stylesheets/lesson-content.css:67]: I feel like these shades of orange are a little too dark when compared to the critical color... What do you think about `@apply border-orange-500 dark:border-orange-400;` instead?

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
This looks awesome! I just have a couple of comments. In addition to the one here, there's a file comment below.

One thing I'm noticing is that the top padding for the note box is a little larger than the bottom padding when there's no heading:

Note box showing larger top padding than bottom padding

Not a major blocker, but it'd be nice if we could update this so that the spacing along the top of the note box is the same as the bottom.

**Comment by luuu-xu** [comment]
> One thing I'm noticing is that the top padding for the note box is a little larger than the bottom padding when there's no heading:

Great eyes!! I have added `-mt-2` to these large icons to compensate the padding and it looks better now.

And I have updated the orange colours for `warning`, they do look much better now. I was too worried about colour contrasts where in fact icon and border does not need to comply strictly to contrast ratings.

**Review by thatblindgeye — APPROVED** [review]
image

😍 

@KevinMulhern if there's nothing you want to look over from a Tailwind perspective this is good to me

File: app/javascript/stylesheets/lesson-content.css [hunk 2]
```diff
@@ -24,33 +24,83 @@
...
+    font-weight: bold;
     content: "\f249";
-    @apply mr-2 text-gold dark:text-gold-700;
+    @apply mr-2 text-gold dark:text-gold-600;
+  }
+
+  .lesson-note > *:only-child::before {
+    display: block;
+    font-family: "Font Awesome 6 Free";
+    font-size: 1.8rem;
```
Comment (KevinMulhern) [inline app/javascript/stylesheets/lesson-content.css:40]: @luuu-xu @thatblindgeye this looks little too big to me in relation to content around it. What do you both think about using something like 1.5? (text-2xl)

1.8
Screenshot 2023-08-09 at 15 06 04

vs

1.5
Screenshot 2023-08-09 at 15 05 46

**Review by KevinMulhern — CHANGES_REQUESTED** [review]

File: app/javascript/stylesheets/lesson-content.css (hunk 2 above)
Comment (thatblindgeye) [reply app/javascript/stylesheets/lesson-content.css:40]: I'm fine with either, as long as the icons are easily recognizable enough. We can always increase/decrease it further if users provide feedback on it

Comment (luuu-xu) [reply app/javascript/stylesheets/lesson-content.css:40]: 1.5 and 1.8 both looks fine to me while 1.5 seems more aesthetically pleasant. Let's do 1.5 first.

**Comment by luuu-xu** [comment]
@KevinMulhern This time I squashed my commits into one commit and rebased onto just pulled upstream main, and then force pushed to my origin fix branch. Please let me know if it looks good and this approach is okay too? I am lost in the ocean of git commands and practices...

**Review by KevinMulhern — APPROVED** [review]
Nice job with rebasing 😎 it's scary at first, but will become second nature in no time.

Thanks for making those changes, the styles look great and the code looks clean ✨ 
Great work!

**Pull request merged**